xcdat/regridder/base.py

```diff
@@ -63,6 +58,40 @@ def _preserve_bounds(
     return output_ds
 
 
+def _drop_axis(ds: xr.Dataset, axis: List[CFAxisKey]) -> xr.Dataset:
+    """Drops an axis or axes in a dataset.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset.
+    axis : List[CFAxisKey]
+        The axis or axes to drop.
+
+    Returns
+    -------
+    xr.Daatset
+        The dataset with axis or axes dropped.
+    """
+    dims: List[str] = []
+
+    for ax in axis:
+        try:
+            dim = get_dim_keys(ds, ax)
+        except KeyError:
+            pass
+        else:
+            if isinstance(dim, str):
+                dims.append(dim)
+            elif isinstance(dim, list):
+                dims = dims + dim
+
+    if len(dims) > 0:
+        ds = ds.drop_dims(dims)
+
+    return ds
```

Comment:
`_drop_axis` calls [get_dim_keys()](https://xcdat.readthedocs.io/en/latest/_modules/xcdat/axis.html#get_dim_keys) to try to get the matching dimension key(s) for an axis. The `ds.drop_dims()` statement is outside of the for loop, which might be more performant than having it inside the loop (did not check).

The tests still pass with this update. Let me know if this is what you were trying to accomplish with the old logic below:

```python
    for x in ignore_dims:
    input_ds = _drop_axis(input_ds, drop_axis)
        try:
            input_ds = input_ds.drop_dims([input_ds.cf[x].name])
        except KeyError:
            to_drop = set(input_ds.cf[[x]].coords.keys()).intersection(
                set(input_ds.coords.keys())
            )
            input_ds = input_ds.drop_dims(list(to_drop))
```